drivers/clock-distribution/ltc6953/ltc6953.c

```diff
@@ -0,0 +1,799 @@
+	case 10:
+		return ltc6953_update(dev, LTC6953_REG(0x05), LTC6953_PD_MSK(channel),
+				      LTC6953_PD(channel, mode));
+	default:
+		return -EINVAL;
+	}
+}
+
+/**
+ * @brief Set LTC6953 Channel Sync mode
+ * @param dev - The device structure.
+ * @param channel - Output channel [0-10]
+ * @param enable - Set to True to enable synchronization mode [SRQMD]
+ * @return Returns 0 in case of success or negative error code.
+ */
+
+int ltc6953_enable_sync(struct ltc6953_dev *dev, int32_t channel, bool enable)
+{
+
+	switch (channel) {
+	case 0:
+		return ltc6953_update(dev, LTC6953_REG(0x0D), LTC6953_SRQEN_MSK,
+				      no_os_field_prep(LTC6953_SRQEN_MSK, (uint8_t)enable));
+	case 1:
+		return ltc6953_update(dev, LTC6953_REG(0x11), LTC6953_SRQEN_MSK,
+				      no_os_field_prep(LTC6953_SRQEN_MSK, (uint8_t)enable));
+	case 2:
+		return ltc6953_update(dev, LTC6953_REG(0x15), LTC6953_SRQEN_MSK,
+				      no_os_field_prep(LTC6953_SRQEN_MSK, (uint8_t)enable));
+	case 3:
+		return ltc6953_update(dev, LTC6953_REG(0x19), LTC6953_SRQEN_MSK,
+				      no_os_field_prep(LTC6953_SRQEN_MSK, (uint8_t)enable));
+	case 4:
+		return ltc6953_update(dev, LTC6953_REG(0x1D), LTC6953_SRQEN_MSK,
+				      no_os_field_prep(LTC6953_SRQEN_MSK, (uint8_t)enable));
+	case 5:
+		return ltc6953_update(dev, LTC6953_REG(0x21), LTC6953_SRQEN_MSK,
+				      no_os_field_prep(LTC6953_SRQEN_MSK, (uint8_t)enable));
+	case 6:
+		return ltc6953_update(dev, LTC6953_REG(0x25), LTC6953_SRQEN_MSK,
+				      no_os_field_prep(LTC6953_SRQEN_MSK, (uint8_t)enable));
+	case 7:
+		return ltc6953_update(dev, LTC6953_REG(0x29), LTC6953_SRQEN_MSK,
+				      no_os_field_prep(LTC6953_SRQEN_MSK, (uint8_t)enable));
+	case 8:
+		return ltc6953_update(dev, LTC6953_REG(0x2D), LTC6953_SRQEN_MSK,
+				      no_os_field_prep(LTC6953_SRQEN_MSK, (uint8_t)enable));
+	case 9:
+		return ltc6953_update(dev, LTC6953_REG(0x31), LTC6953_SRQEN_MSK,
+				      no_os_field_prep(LTC6953_SRQEN_MSK, (uint8_t)enable));
+	case 10:
+		return ltc6953_update(dev, LTC6953_REG(0x35), LTC6953_SRQEN_MSK,
+				      no_os_field_prep(LTC6953_SRQEN_MSK, (uint8_t)enable));
+	default:
+		return -EINVAL;
+	}
+
+
+}
+
+/**
+ * @brief Set digital delay/phase of LTC6953 output channel
+ * @param dev - The device structure.
+ * @param channel - Output channel [0-10]
+ * @param delay - 16-bit word for delay value
+ * @return Returns 0 in case of success or negative error code.
+ */
+
+int ltc6953_set_digital_delay (struct ltc6953_dev *dev, int32_t channel,
+			       uint16_t delay)
+{
+	int ret;
+	uint16_t ddel_high;
+	uint16_t ddel_low;
+
+	if (delay > 4095)
+		return -EINVAL;
+
+	ddel_low  = (uint8_t)no_os_field_get(LTC6953_DDEL_LOW_MSK, delay);
+	ddel_high = (uint8_t)no_os_field_get(LTC6953_DDEL_HIGH_MSK,
+					     no_os_shift_right(delay, 8));
+
+	switch (channel) {
+	case 0:
+		ret = ltc6953_update(dev, LTC6953_REG(0x0D), LTC6953_DDEL_HIGH_MSK,
+				     no_os_field_prep(LTC6953_DDEL_HIGH_MSK, ddel_high));
+		if (ret)
```

Comment:
I think this should be if (!ret) since 0 is the successful result. It would only proceed with the other register update if the first one is successful.